aptos-move/parallel-executor/src/executor.rs

```diff
@@ -125,6 +125,15 @@ impl<
                         None => continue,
                     }
                 }
+                Err(DeltaApplicationFailure) => {
+                    // Delta application failure currently should never happen. Here, we assume it
+                    // happened because of speculation and return 0 to the Move-VM. Validation will
+                    // ensure the transaction re-executes if 0 wasn't the right number.
+                    self.captured_reads
+                        .lock()
+                        .push(ReadDescriptor::from_delta_application_failure(key.clone()));
```

Comment:
Isn't that the intended behavior, we wanted to panic instead of spinning forever if the assumption is violated, and if it is supposed to read a bad number, it will pass validation and fail with panic when outputs are prepared.

Reply:
Why would it fail when output are prepared? Are we reading read descriptions at that point?